File: Backend/utils/common/responses.js
export const buildApiResponse = ({
  success,
  message,
  data = null,
  error = null,
}) => {
  if (success) {
    return {
      success: true,
      message,
      data: data ?? null,
    };
  }

  return {
    success: false,
    message,
    data: null,
    error: error ?? null,
  };
};

export const sendSuccess = (res, message, data = null, statusCode = 200) => {
  return res
    .status(statusCode)
    .json(
      buildApiResponse({
        success: true,
        message,
        data,
      })
    );
};

File: ApiModels/services/model_executors/responses.py
from typing import Any

from fastapi.responses import JSONResponse


def success_response(message: str, data: Any) -> dict[str, Any]:
    return {
        "success": True,
        "message": message,
        "data": data,
    }


def error_response(
    message: str,
    code: str = "MODEL_EXECUTION_ERROR",
    details: Any | None = None,
) -> JSONResponse:
    return JSONResponse(
        status_code=200,
        content={
            "success": False,
            "message": message,
            "data": None,
            "error": {
                "code": code,
                "field": None,
                "details": details,
            },
        },
    )
